src/api.h

```diff
@@ -57,6 +60,8 @@ AWS_CRT_API const char *aws_crt_error_name(int);
 AWS_CRT_API const char *aws_crt_error_debug_str(int);
 AWS_CRT_API void aws_crt_reset_error(void);
 
+AWS_CRT_API void aws_crt_thread_join_all(void);
```

Comment:
In C, this function raises an error if the threads didn't manage to join within the desired timeout (controlled by `aws_thread_set_managed_join_timeout_ns(timeout)`).

The default timeout is "wait forever". So if any of your tests are keeping a thread alive, CI will just freeze when this is called until Github kills it. Your probably want to:
1) Expose the set_timeout() function
2) have join_all() return `int` so callers can tell whether or not it succeeded
3) Have your tests set the timeout to something reasonable like 10sec so they don't hang forever
4) Have your tests check the return code of join_all()

Reply:
Can do.